src/cmd/troubleshoot/dynakube.go

```diff
@@ -0,0 +1,195 @@
+package troubleshoot
+
+import (
+	"context"
+	"fmt"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/dtpullsecret"
+	"github.com/Dynatrace/dynatrace-operator/src/dtclient"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/Dynatrace/dynatrace-operator/src/webhook/validation"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	pullSecretFieldValue = "top-secret"
+)
+
+func checkDynakube(troubleshootCtx *troubleshootContext) error {
+	log = newTroubleshootLogger("[dynakube  ] ")
+
+	logNewTestf("checking if '%s:%s' Dynakube is configured correctly", troubleshootCtx.namespaceName, troubleshootCtx.dynakubeName)
+
+	tests := []troubleshootFunc{
+		checkDynakubeCrdExists,
+		getSelectedDynakubeIfItExists,
+		checkApiUrl,
+		evaluateDynatraceApiSecretName,
+		getDynatraceApiSecretIfItExists,
+		checkIfDynatraceApiSecretHasApiToken,
+		evaluatePullSecret,
+		getPullSecretIfItExists,
+		checkPullSecretHasRequiredTokens,
+		evaluateProxySecret,
+		getProxySecretIfItExists,
+		checkProxySecretHasRequiredTokens,
+	}
+
+	for _, test := range tests {
+		if err := test(troubleshootCtx); err != nil {
+			logErrorf(err.Error())
+			return fmt.Errorf("'%s:%s' Dynakube isn't valid", troubleshootCtx.namespaceName, troubleshootCtx.dynakubeName)
+		}
+	}
+
+	logOkf("'%s:%s' Dynakube is valid", troubleshootCtx.namespaceName, troubleshootCtx.dynakubeName)
+	return nil
+}
+
+func checkDynakubeCrdExists(troubleshootCtx *troubleshootContext) error {
+	dynakubeList := &dynatracev1beta1.DynaKubeList{}
+	if err := troubleshootCtx.apiReader.List(context.TODO(), dynakubeList, &client.ListOptions{Namespace: troubleshootCtx.namespaceName}); err != nil {
+		return errorWithMessagef(err, "CRD for Dynakube missing")
+	}
+	logInfof("CRD for Dynakube exists")
+	return nil
+}
+
+func getSelectedDynakubeIfItExists(troubleshootCtx *troubleshootContext) error {
+	query := kubeobjects.NewDynakubeQuery(troubleshootCtx.apiReader, troubleshootCtx.namespaceName).WithContext(context.TODO())
+	if dynakube, err := query.Get(types.NamespacedName{Namespace: troubleshootCtx.namespaceName, Name: troubleshootCtx.dynakubeName}); err != nil {
+		return errorWithMessagef(err, "selected '%s:%s' Dynakube does not exist", troubleshootCtx.namespaceName, troubleshootCtx.dynakubeName)
+	} else {
+		troubleshootCtx.dynakube = dynakube
+	}
```

Comment:
```suggestion
	}
	troubleshootCtx.dynakube = dynakube
```
we won't reach this point if there's an error